pkg/bgpv1/manager/reconcilerv2/policies_test.go

```diff
@@ -0,0 +1,479 @@
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+)
+
+func Test_MergeRoutePolicies(t *testing.T) {
+
+	localPrefDefault := int64(100)
+	localPrefLow := int64(50)
+	localPrefHigh := int64(200)
+
+	tests := []struct {
+		name          string
+		policyA       *types.RoutePolicy
+		policyB       *types.RoutePolicy
+		errorExpected bool
+		expected      *types.RoutePolicy
+	}{
+		{
+			name:          "nil policy - both",
+			policyA:       nil,
+			policyB:       nil,
+			errorExpected: true,
+			expected:      nil,
+		},
+		{
+			name:    "nil policy - policyA",
+			policyA: nil,
+			policyB: &types.RoutePolicy{
+				Name: "policy",
+			},
+			errorExpected: true,
+			expected:      nil,
+		},
+		{
+			name: "nil policy - policyB",
+			policyA: &types.RoutePolicy{
+				Name: "policy",
+			},
+			policyB:       nil,
+			errorExpected: true,
+			expected:      nil,
+		},
+		{
+			name: "policy names mismatched",
+			policyA: &types.RoutePolicy{
+				Name: "policy",
+			},
+			policyB: &types.RoutePolicy{
+				Name: "notpolicy",
+			},
+			errorExpected: true,
+			expected:      nil,
+		},
+		{
+			name: "policy types mismatched",
+			policyA: &types.RoutePolicy{
+				Name: "policy",
+				Type: types.RoutePolicyTypeExport,
+			},
+			policyB: &types.RoutePolicy{
+				Name: "policy",
+				Type: types.RoutePolicyTypeImport,
+			},
+			errorExpected: true,
+			expected:      nil,
+		},
+		{
+			name: "empty statements",
+			policyA: &types.RoutePolicy{
+				Name:       "policy",
+				Statements: []*types.RoutePolicyStatement{},
+			},
+			policyB: &types.RoutePolicy{
+				Name:       "policy",
+				Statements: []*types.RoutePolicyStatement{},
+			},
+			errorExpected: false,
+			expected: &types.RoutePolicy{
+				Name:       "policy",
+				Statements: []*types.RoutePolicyStatement{},
+			},
+		},
+		{
+			name: "nil statement.Action.SetLocalPref",
+			policyA: &types.RoutePolicy{
+				Name: "policy",
+				Statements: []*types.RoutePolicyStatement{
+					{
+						Conditions: types.RoutePolicyConditions{},
+						Actions: types.RoutePolicyActions{
+							RouteAction:        types.RoutePolicyActionAccept,
+							SetLocalPreference: nil,
+						},
+					},
+				},
+			},
+			policyB: &types.RoutePolicy{
+				Name: "policy",
+				Statements: []*types.RoutePolicyStatement{
+					{
+						Conditions: types.RoutePolicyConditions{},
+						Actions: types.RoutePolicyActions{
+							RouteAction:        types.RoutePolicyActionAccept,
+							SetLocalPreference: nil,
+						},
+					},
+				},
+			},
+			errorExpected: false,
+			expected: &types.RoutePolicy{
+				Name: "policy",
+				Statements: []*types.RoutePolicyStatement{
+					{
+						Conditions: types.RoutePolicyConditions{},
+						Actions: types.RoutePolicyActions{
+							RouteAction:        types.RoutePolicyActionAccept,
+							SetLocalPreference: nil,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "both set standard communities",
```

Comment:
These tests could use another round of polish.  I intend to refactor this a bit more tomorrow using variables for some of the duplicate blocks.